### PR TITLE
Update `CupertinoPageRoute` transition animation curves

### LINKED ISSUE #122247: Transition animation curves used by Cupertino are outdated
CupertinoPageRoute animations are now outdated and don't feel native to the user.
Related to the https://github.com/flutter/flutter/issues/22862

**Expected results:**
Transition animation being close to native.

**Actual results:**
1. Transition duration is 100 ms shorter than native iOS.
2. Native transition starts slowly and quickly accelerates, rather than being linear from the beginning.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/cupertino.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return CupertinoApp(
      title: 'Cupertino App',
      home: CupertinoPageScaffold(
        navigationBar: const CupertinoNavigationBar(
          middle: Text('Cupertino App Bar'),
        ),
        child: Center(
          child: Builder(builder: (context) {
            return CupertinoButton(
              child: const Text('Next Page'),
              onPressed: () {
                Navigator.push(
                  context,
                  CupertinoPageRoute(builder: (context) => const SecondPage()),
                );
              },
            );
          }),
        ),
      ),
    );
  }
}

class SecondPage extends StatelessWidget {
  const SecondPage({super.key});

  @override
  Widget build(BuildContext context) {
    return CupertinoPageScaffold(
      navigationBar: const CupertinoNavigationBar(
        middle: Text('Second Page'),
      ),
      child: Center(
        child: CupertinoButton(
          child: const Text('Back'),
          onPressed: () {
            Navigator.pop(context);
          },
        ),
      ),
    );
  }
}

```
</details>

https://user-images.githubusercontent.com/57679865/223876857-16f78682-9b24-4d32-9daf-d299efd57c03.mov

### PR DESCRIPTION
**Pull request by ivirtex**
This PR updates transition animation curves and transition duration used by `CupertinoPageRoute`.
Docs part: https://github.com/flutter/assets-for-api-docs/pull/211

Fixes https://github.com/flutter/flutter/issues/122247

| Before  | After |
| ------------- | ------------- |
| <video src="https://user-images.githubusercontent.com/57679865/226168675-2d24d056-91d9-498f-9808-ee08d240c293.mov"> | <video src="https://user-images.githubusercontent.com/57679865/226168431-00b60ea9-33f4-4458-ae62-069db66c6ff7.mov">   |

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by ivirtex** [comment]
cc @MitchellGoodwin

**Review by MitchellGoodwin — APPROVED** [review]
LGTM

**Comment by ivirtex** [comment]
@MitchellGoodwin I've put this curve in the Curves class, but I'm not sure how to generate animation previews as other Curves have in their doc comments e.g.:
`/// {@animation 464 192 https://flutter.github.io/assets-for-api-docs/assets/animation/curve_fast_linear_to_slow_ease_in.mp4}`

https://user-images.githubusercontent.com/57679865/226567852-32b5df0c-1177-4cc0-94c8-646bd1b2b62e.mp4

**Comment by ivirtex** [comment]
> I believe that's done through this repo: https://github.com/flutter/assets-for-api-docs

Thanks! I've just created a PR on this repo: https://github.com/flutter/assets-for-api-docs/pull/211 and added link to relevant diagram in the doc comments on this side.

**Review by MitchellGoodwin — APPROVED** [review]
LGTM. The other PR for the diagram also looks good to me as well. I think this one can be merged as is first, and the other will have to wait until this code is live in stable. Feel free to ping me if I don't get back to this to merge or comment on this by next week.

**Review by MitchellGoodwin — APPROVED** [review]

**Comment by ivirtex** [comment]
@MitchellGoodwin Can we merge this PR manually or should we wait until the bug with auto-submit bot is fixed?

File: packages/flutter/lib/src/animation/curves.dart [hunk 1]
```diff
@@ -1385,6 +1385,24 @@ class Curves {
   /// {@animation 464 192 https://flutter.github.io/assets-for-api-docs/assets/animation/curve_fast_linear_to_slow_ease_in.mp4}
   static const Cubic fastLinearToSlowEaseIn = Cubic(0.18, 1.0, 0.04, 1.0);
 
+  /// A curve that starts slowly, speeds up very quickly, and then ends slowly.
+  ///
+  /// This curve is used by default to animate page transitions used by CupertinoPageRoute.
```
Comment (justinmc) [inline packages/flutter/lib/src/animation/curves.dart:1390]: Link it: `[CupertinoPageRoute]`.

Also, maybe it should be on the next line (but it's hard to tell on GitHub).

File: dev/integration_tests/flutter_gallery/test/demo/cupertino/cupertino_navigation_demo_test.dart [hunk 2]
```diff
@@ -32,7 +32,7 @@ void main() {
     // Tap some row to go to the next page.
     await tester.tap(find.text('Buy this cool color').first);
     await tester.pump();
-    await tester.pump(const Duration(milliseconds: 500));
+    await tester.pump(const Duration(milliseconds: 600));
```
Comment (justinmc) [inline dev/integration_tests/flutter_gallery/test/demo/cupertino/cupertino_navigation_demo_test.dart:35]: Is this based on CupertinoRouteTransitionMixin.transitionDuration?  I wonder if you could base it on that.  No big deal if not, though.

**Review by justinmc — APPROVED** [review]
LGTM 👍 

This looks slick, thank you for fixing it!

**Pull request merged**